lib/household/view/view_household.dart

```diff
@@ -0,0 +1,157 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:gibsonify/household/household.dart';
+import 'package:intl/intl.dart';
+
+import '../../navigation/models/page_router.dart';
+
+class ViewHouseholdPage extends StatelessWidget {
+  const ViewHouseholdPage({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocBuilder<HouseholdBloc, HouseholdState>(
+      builder: (context, state) {
+        return Scaffold(
+          appBar: AppBar(
+            leading: IconButton(
+                onPressed: () {
+                  Navigator.of(context, rootNavigator: true).pop();
+                },
+                icon: const Icon(Icons.arrow_back)),
+            title: state is HouseholdLoaded
+                ? Text(state.household!.householdId)
+                : const Text('Household'),
+            actions: state is HouseholdLoaded
+                ? [
+                    IconButton(
+                        onPressed: () => {
+                              Navigator.pushNamed(
+                                context,
+                                PageRouter.editHousehold,
+                              )
+                            },
+                        icon: const Icon(Icons.edit))
+                  ]
+                : [],
+          ),
+          body: Padding(
+            padding: const EdgeInsets.all(8.0),
+            child: householdView(context, state),
+          ),
+        );
+      },
+    );
+  }
+}
+
+Widget householdView(context, state) {
```

Comment:
can this be put inside the viewhousehold page class as its only used there

Reply:
it can be placed inside the build context and then you don't need to pass context in too

Reply:
I've made it a method because I couldn't figure out a way to write it in-line while having the `if..else` switch based on the state. Is there a way other than using a lot of ternary operators?

Reply:
Keep it as a method just place it inside the Build context of the above widget so you don't need to pass in context

Reply:
Done